### ISSUE TITLE
[BITV] Enhance Multiselect A11Y

### ISSUE THREAD

**Issue by Pytal** [nextcloud/nextcloud-vue#3526]
- [x] aria-label needs to be removed for unclickable avatars https://github.com/nextcloud/nextcloud-vue/pull/3618
- ~https://github.com/sagalbot/vue-select/pull/1726~ Using fork https://github.com/nextcloud-deps/vue-select
- ~https://github.com/sagalbot/vue-select/pull/1742~ Using fork https://github.com/nextcloud-deps/vue-select
- [x] https://github.com/nextcloud-deps/vue-select/pull/1
- [x] https://github.com/nextcloud-deps/vue-select/pull/2
- [x] https://github.com/nextcloud-deps/vue-select/pull/8
- [x] https://github.com/nextcloud/nextcloud-vue/pull/3703
- [x] https://github.com/nextcloud-deps/vue-select/pull/9
- [x] https://github.com/nextcloud-deps/vue-select/pull/10
- [x] https://github.com/nextcloud-deps/vue-select/pull/11
- [x] https://github.com/nextcloud/nextcloud-vue/pull/3755
- [x] https://github.com/nextcloud/server/pull/36813
- [x] https://github.com/nextcloud/server/pull/36841

**Comment by Pytal** [comment]
@michaelnissenbaum when there are potentially many options to pick from in a multiselect dropdown do we need to add pagination like in https://vue-select.org/guide/pagination.html? Would the user need to be able to jump to a specific page i.e. page #5 out of 10 pages?

Or is making the option list filterable by typing sufficient to pass the accessibility audit?

**Comment by michaelnissenbaum** [comment]
Which kind of pagination navigation you select/implement is up to you. "all" of them could be implemented in accessible way. If you have already something in mind let's discuss whether it is accessible or not and if not, what must be done. Provide me some screenshots and HTML code and I'd analyse it and provide you my feedback.

**Comment by Pytal** [comment]
Currently we do not have any pagination but do have filtering

Our current implementation is available to test in https://nextcloud-vue-components.netlify.app/#/Components/NcSelect?id=ncselect-1, you may scroll down to the "User select examples" section specifically

Feel free to test it out by typing to filter the available options and let us know whether this is sufficient to pass the audit? Or if adding some buttons similar to https://vue-select.org/guide/pagination.html 
![image](https://user-images.githubusercontent.com/24800714/205186338-30b15493-cdbc-4413-977f-837fac6b6389.png)
would be necessary?

**Comment by michaelnissenbaum** [comment]
I took a look at those examples. What I can say - they are not accessible.
1. ARIA roles/attributes were not used as expected. Screen reader does not provide expected/required information. A working example could be found at https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-autocomplete-both.html.
2. Combobox itself (border) does not meet contrast requirements
3. Keyboard focus (within listbox) does not meet contrast requirements

Regarding multi select combobox take a look at this example at http://accessibleculture.org/misc/aria/combobox/combobox.html. At least it works quite well with screen reader. Additionally take a loot at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-multiselectable. The current vue version for multi select does not really support screen readers. It provides incorrect information to the users.

**Comment by Pytal** [comment]
For the multi select combobox specifically, would the current structure
![image](https://user-images.githubusercontent.com/24800714/206334977-4c8383ad-537d-4d1b-9f5c-1828a7936d2d.png)
with the non-comma separated selected options display and without the per-option checkboxes seen below,
![image](https://user-images.githubusercontent.com/24800714/206335086-607571f8-c37a-4df9-978c-04c6e87a335f.png)
be accessible if the ARIA and contrast adjustments are made but without the same structure?

**Comment by michaelnissenbaum** [comment]
The implementation without checkboxes could be an accessible one in case proper ARIA roles/attributes are used, text and keyboard focus meet contrast requirements and keyboard support is provided. An example could be found at https://www.telerik.com/kendo-angular-ui/components/dropdowns/multiselect/accessibility/.

**Comment by Pytal** [comment]
What contrast ratio is required for the focused dropdown option @michaelnissenbaum?
![image](https://user-images.githubusercontent.com/24800714/206334977-4c8383ad-537d-4d1b-9f5c-1828a7936d2d.png)

**Comment by michaelnissenbaum** [comment]
The contrast ration between focused and unfocussed state should be at least 3:1. Please don't forget that in case that keyboard focus is implemented using background colour change the contrast ration between text and its background has to be at least 4,5:1.

**Comment by Pytal** [comment]
Would something like this work?
![image](https://user-images.githubusercontent.com/24800714/207990793-70f05e94-e44f-4aec-9c85-7338638f1896.png)

**Comment by michaelnissenbaum** [comment]
I guess you mean keyboard focus borders. It would work but only in case the contrast ratio between border and its background is at least 3:1. At the moment we have appr. 2:1.
![CleanShot 2022-12-16 at 08 34 35@2x](https://user-images.githubusercontent.com/48948959/208047044-1fcf7047-2267-4c66-b55f-d069352f45e9.png)

**Comment by Pytal** [comment]
Transferred "Focus of search suggestions is not clearly visible, needs border instead of background"
- https://github.com/nextcloud/nextcloud-vue/issues/3642
